### ISSUE TITLE
Edge : deselectEdge not fired

### ISSUE THREAD

**Issue by jk-12345** [visjs/vis-network#1277]
Hi,

The current version of visjs (and previous ones ?) doesn't fire deselectEdge once we unselect an edge.

U can check out the online exemples https://visjs.github.io/vis-network/examples/network/events/interactionEvents.html : even if deselectEdge is "on", no event is reported to the console.

Thanks for helping

**Comment by mliu49** [comment]
I also encountered this issue. After some lengthy debugging, I discovered that it's due to #1200 (c5aba688a17be664be0e454de0fadb97ad86b8e0). On click events, the selection changes used to be committed by `commitAndEmit()` [here](https://github.com/visjs/vis-network/blob/master/lib/network/modules/InteractionHandler.js#L155). Now, they are committed early when `unselectAll()` is called, so the `commitAndEmit()` function is unable to detect deselect events.

Unfortunately, I don't know what the appropriate fix is, since the change was introduced to fix a different bug. Hopefully someone who's more familiar with this can help out.
